omnipaxos_storage/src/persistent_storage.rs

```diff
@@ -199,8 +203,10 @@ where
     }
 
     fn batch_append_on_prefix(&mut self, from_idx: usize, entries: Vec<T>) -> StorageResult<()> {
-        if from_idx < self.next_log_key {
-            let from_key = from_idx.to_be_bytes();
+        // Don't need to delete entries will we overwrite.
```

Comment:
I don't see where the entries get overwritten? Here we only delete the entries that get dropped (which is correct), but we also need to overwrite entries that are between from_idx and delete_idx